### PR TITLE
Add shared and static flags to diligent core

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
Specify library name and version:  **diligent-core/2.5.1**

introduce shared flag for diligent-core.
The lib always produce two sets of libs: static and dynamic. All we need, is to delete another set in package method

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by AndreyMlashkin** [comment]
I have this line in recipe, isn't it enough?
    def config_options(self):
        if self.settings.os == "Windows" or self.options.shared:
            del self.options.fPIC
            
 @ericLemanissierBot @prince-chrismc @SSE4

**Comment by SSE4** [comment]
> I have this line in recipe, isn't it enough? def config_options(self): if self.settings.os == "Windows" or self.options.shared: del self.options.fPIC
> 
> @ericLemanissierBot @prince-chrismc @SSE4

try the following code block:
```
    def configure(self):
        if self.options.shared:
            del self.options.fPIC

    def config_options(self):
        if self.settings.os == "Windows":
            del self.options.fPIC
```
it's proven to pass hooks.
I know it might be a bit annoying, and hook sometimes might be too strict rejecting a valid code.

**Comment by AndreyMlashkin** [comment]
How conan handels shared flag internally?
We don't need this flag here, because Diligent produces two sets of libs - shared and static and conan should just package relevant ones.
How can I drop the 'shared' flag in cmake?

@uilianries @SSE4 @ericLemanissierBot @jgsogo

**Comment by uilianries** [comment]
@AndreyMlashkin Please, read https://docs.conan.io/en/latest/reference/build_helpers/cmake.html#definitions

**Comment by SSE4** [comment]
> How conan handels shared flag internally? We don't need this flag here, because Diligent produces two sets of libs - shared and static and conan should just package relevant ones. How can I drop the 'shared' flag in cmake?

also, might be relevant for diligent case: https://docs.conan.io/en/1.44/reference/conanfile/methods.html#build-id

**Comment by AndreyMlashkin** [comment]
> @AndreyMlashkin Please, read https://docs.conan.io/en/latest/reference/build_helpers/cmake.html#definitions

I will rephrase my question: how can I drop BUILD_SHARED_LIBS flag from self._cmake?

**Comment by uilianries** [comment]
Still, we only package only one set per package, you don't need to build both, it will consume more build time only.

**Comment by AndreyMlashkin** [comment]
> Still, we only package only one set per package, you don't need to build both, it will consume more build time only.

if I override build_id, will it be possible to have one build directory and two different package directories? As far as I know, conan does not have such logic

**Comment by AndreyMlashkin** [comment]
I need to disable spv_remapper lib

**Comment by AndreyMlashkin** [comment]
I need to link against glslang::glslang and not against glslang::spvremapper, what I am doing wrong here?...

**Comment by AndreyMlashkin** [comment]
@uilianries @SSE4 @jgsogo @SpaceIm @prince-chrismc 
ping

File: recipes/diligent-core/all/conanfile.py [hunk 1]
```diff
@@ -148,11 +153,19 @@ def package(self):
...
-        self.copy(pattern="*.lib", dst="lib", keep_path=False)
-        self.copy(pattern="*.dylib", dst="lib", keep_path=False)
-        self.copy(pattern="*.so", dst="lib", keep_path=False)
-        self.copy(pattern="*.dll", dst="bin", keep_path=False)
+        if self.options.shared:
+            self.copy(pattern="*.dylib", dst="lib", keep_path=False)
+            self.copy(pattern="*.so", dst="lib", keep_path=False)
+            self.copy(pattern="*.dll", dst="bin", keep_path=False)
+            tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.a")
+            tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.lib")
```
Comment (uilianries) [inline recipes/diligent-core/all/conanfile.py:161]: on Windows you still need .lib with the exported symbols, otherwise, will need to load .dll manually.

**Review by uilianries — APPROVED** [review]

**Comment by AndreyMlashkin** [comment]
@SSE4 @jgsogo @SpaceIm @prince-chrismc
please approve

**Comment by AndreyMlashkin** [comment]
@SSE4 @jgsogo @SpaceIm @prince-chrismc @danimtb 
please approve

**Review by jgsogo — APPROVED** [review]

**Pull request merged**